app/services/catalog/update_order_item.rb

```diff
@@ -31,13 +31,12 @@ def find_order_item
 
     def fetch_external_url
       TopologicalInventory.call do |api_instance|
-        @service_instance_id = @task.context[:service_instance][:id]
-        service_instance = api_instance.show_service_instance(@service_instance_id)
+        service_instance = api_instance.show_service_instance(service_instance_id)
         raise ServiceInstanceWithoutExternalUrl if service_instance.external_url.nil?
```

Comment:
I realize this was here before these changes, but are we using the `raise` to perform control flow which is normally considered a bad practice.

It actually feels like the raise is making the logic of logging the lack of external_url more difficult.  I think the following would give the same result:
```ruby
def fetch_external_url
  TopologicalInventory.call do |api_instance|
    service_instance = api_instance.show_service_instance(service_instance_id)
    if service_instance.external_url.nil?
      Rails.logger.error("Could not find an external url on service instance (id: #{service_instance_id}) attached to task_id: #{@payload["task_id"]}")
    end
    service_instance.external_url
  end
end
```

And is this really an error condition or a warning?

Reply:
It looks like its just a warning, updated it to be more clear. `external_url` will be `nil` so no need to rescue and return `nil`.

Reply:
@lindgrenj6 If we think this is a warning I would also change the `Rails.logger.error` to log as a warning as well.

Reply:
OH, yeah missed that. Thanks.

Reply:
Sorry, last thing then I can approve.  We no longer need the `ServiceInstanceWithoutExternalUrl` error class defined at the top of this file.  I should have caught that before.

Reply:
Glad you're being diligent, because apparently my diligence level is not high enough today!